src/components/FieldsetItems/FieldsetItems.tsx

```diff
@@ -26,10 +26,9 @@ export type FieldsetItemsProps = {
  */
 export const FieldsetItems = ({
   children,
-  as: Component = 'div',
+  as: Component = 'div' as const,
   className,
 }: FieldsetItemsProps) => {
   const componentClassName = clsx(styles['fieldset-items'], className);
-  // @ts-expect-error TODO: investigate error
   return <Component className={componentClassName}>{children}</Component>;
```

Comment:
What was the error here that typescript was flagging?

Although the error was removed, I think the type safety is slightly worse now. Now that the type is `as?: ElementType | string;` a developer could accidentally type `<FieldsetItems as="spna" />` and TS wouldn't complain

Reply:
@jeremiah-clothier I left off the links as they point to internal trackers in this case, but would generally prefer linking them. Relevant tickets are linked to the ticket name in the branch, for safe keeping.

Reply:
oops, responded to the wrong comment :/ @jeremiah-clothier that's a good point actually. I saw something similar for our other `as` props, but this could be a problem. I'll double check (just in case react has some internal handling for this param when converting JSX to actual function calls. What I want here is "strings that can be used as the list of supported component names"

Reply:
@jeremiah-clothier here's the full error once i remove the `string` type:

```
'Component' cannot be used as a JSX component.
  Its element type 'Element | Component<any, any, any> | null' is not a valid JSX element.
    Type 'Component<any, any, any>' is not assignable to type 'Element | ElementClass | null'.
      Type 'Component<any, any, any>' is not assignable to type 'ElementClass'.
        The types returned by 'render()' are incompatible between these types.
          Type 'React.ReactNode' is not assignable to type 'import("/Users/aholloway/Development/edu-design-system/node_modules/@types/react-beautiful-dnd/node_modules/@types/react/index").ReactNode'.
            Type '{}' is not assignable to type 'ReactNode'.ts(2786)
```
I have a few ideas how to resolve this but will do some more digging tomorrow. I'm not sure where that `null` is coming from, which I think is the main issue.

Reply:
@jeremiah-clothier added a comment for this, and fixed up the types. unfortunately looks like we may need to keep this expected error for a while longer :( 

`Type 'React.ReactNode' is not assignable to type 'import("/Users/aholloway/Development/edu-design-system/node_modules/@types/react-beautiful-dnd/node_modules/@types/react/index").ReactNode'.` is coming from a conflict between our version of react, and one that this dependency is pulling in via its own node module code :/